doc/source/ray-air/examples/batch_tuning.ipynb

```diff
@@ -0,0 +1,1197 @@
+    "- Since we are using **grid search**, this means `train_model()` will be run *in parallel for every permutation* in the Tune search space!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "a03c0ff8-0c42-4703-91f4-57fc0df8f0ea",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 2. Define a custom train function\n",
+    "def train_model(config: dict):\n",
+    "\n",
+    "    model = config['model']\n",
+    "    the_location = config['location']\n",
+    "    \n",
+    "    # Load data.\n",
+    "    df_list = [read_data(f, the_location) for f in s3_files]   \n",
+    "    df_raw = pd.concat(df_list, ignore_index=True)\n",
+    "    df = transform_df(df_raw)\n",
+    "    \n",
+    "    # Train/valid split.\n",
+    "    train_df, valid_df = train_test_split(df, test_size=0.2, shuffle=True)\n",
+    "    train_X = train_df[[\"passenger_count\", \"trip_distance\"]]\n",
+    "    train_y = train_df.trip_duration\n",
+    "    valid_X = valid_df[[\"passenger_count\", \"trip_distance\"]]\n",
+    "    valid_y = valid_df.trip_duration\n",
+    "\n",
+    "    # Train model.\n",
+    "    model = model.fit(train_X, train_y)\n",
+    "    pred_y = model.predict(valid_X)\n",
+    "    \n",
+    "    # Evaluate.\n",
+    "    error = sklearn.metrics.mean_absolute_error(valid_y, pred_y)\n",
+    "    \n",
+    "    # Save the model as a Tune Checkpoint.  \n",
+    "    # This is done through ray.air.session.report() API.\n",
+    "    # https://docs.ray.io/en/latest/tune/tutorials/tune-checkpoints.html\n",
+    "    \n",
+    "    # Define a model checkpoint.\n",
+    "    checkpoint = SklearnCheckpoint.from_estimator(model, path=\".\")\n",
+    "\n",
+    "    # Save checkpoint and report back metrics, using ray.air.session.report()\n",
+    "    metrics = dict(error = error)\n",
+    "    session.report(\n",
+    "            metrics, \n",
+    "            checkpoint=checkpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ebb435c-8485-4146-9e22-cff77b3f4946",
+   "metadata": {},
+   "source": [
+    "## Run independent experiments in Parallel using Ray Tune <a class=\"anchor\" id=\"run_tune_search\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "222d0d25-a5b0-4887-9d1a-a775d0107d12",
+   "metadata": {},
+   "source": [
+    "Now we are ready to use Ray Tune to run separate, parallel training jobs, each as a different experiment, per dropoff location.\n",
+    "\n",
+    "**Configure the resources allocated per experiment (trial).** Tune uses this resources allocation to control the parallelism. For example, if each trial was configured to use 4 CPUs, and the cluster had only 32 CPUs, then Tune will limit the number of concurrent trials to 8 to avoid overloading the cluster. For more information, see [A Guide To Parallelism and Resources](https://docs.ray.io/en/master/tune/tutorials/tune-resources.html#tune-parallelism)."
```

Comment:
```suggestion
    "**Configure the resources allocated per trial.** Tune uses this resources allocation to control the parallelism. For example, if each trial was configured to use 4 CPUs, and the cluster had only 32 CPUs, then Tune will limit the number of concurrent trials to 8 to avoid overloading the cluster. For more information, see [A Guide To Parallelism and Resources](https://docs.ray.io/en/master/tune/tutorials/tune-resources.html#tune-parallelism)."
```